### PR TITLE
DeepONet Example

### PR DESCRIPTION
**Pull request by colbyham**
First DeepONet example, wrapping up modifications in this new branch after messy rebase

### CODE REVIEW

**Comment by RBirmiwal** [comment]
This is looking good so far!

**Review by drgona — CHANGES_REQUESTED** [review]
I like the new DeepONet class abstraction 
- lets add explanation in the docstring on problem dimensions: Nu, Nsamples, in_size_trunk, interact_size
- Lets try to make the example more didactical. We should add a paragraph explaining DeepONet architecture including figure from the paper.
- we need to add more markdown in the code - each cell should have a brief markdown explanation 
- lets avoid the use of undefined acronyms - for instance, its not clear what is 1d grf from deepxde - moreover as far as I can tell these features are not used further in the code - each cell needs to be intentional. We should avoid junk code in the example
- lets avoid cells that generate more than single figure 
- **IMPORTANT**: current example will not run - it expects training data from examples/neural_operators/datasets - Lets generate data by simulating the system not by loading from any file
- I don't see the utility of cell with scipy integrate and associated plot
- I don't see the need for changes in Trainer and Problem classes currently suggested

**Comment by colby-ham** [comment]
Will briefly respond to the comments before our meeting tomorrow. The "1d grf from deepxde" is part of me trying to get the data generation working in the example and that is also the reason why multiple plots are being show for my debugging purposes. I talked to @brunopjacob and think I have a way forward for data generation. The extra changes from Trainer and Problem are accidental from code reformat. I will pull a fresh copy and only put the changes necessary Trainer ( loss history), Problem (predict)

**Comment by RBirmiwal** [comment]
@brunopjacob Let's plan to finish this this week

**Comment by Parv621** [comment]
@drgona This PR can be deleted as #265 covers it fully
